src/spatialdata/_core/operations/aggregate.py

```diff
@@ -0,0 +1,261 @@
+from __future__ import annotations
+
+from typing import Any
+
+import anndata as ad
+import dask as da
+import dask.dataframe as ddf
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+from multiscale_spatial_image import MultiscaleSpatialImage
+from scipy import sparse
+from spatial_image import SpatialImage
+from xrspatial import zonal_stats
+
+from spatialdata._core.operations.transform import transform
+from spatialdata._types import ArrayLike
+from spatialdata.models import (
+    Image2DModel,
+    Labels2DModel,
+    PointsModel,
+    ShapesModel,
+    get_model,
+)
+from spatialdata.models._utils import get_axes_names
+from spatialdata.transformations import BaseTransformation, Identity, get_transformation
+
+__all__ = ["aggregate"]
+
+
+def aggregate(
+    values: ddf.DataFrame | gpd.GeoDataFrame | SpatialImage | MultiscaleSpatialImage,
+    by: gpd.GeoDataFrame | SpatialImage | MultiscaleSpatialImage,
+    id_key: str | None = None,
+    *,
+    value_key: str | None = None,
+    agg_func: str | list[str] = "mean",
+    target_coordinate_system: str = "global",
+    **kwargs: Any,
+) -> ad.AnnData:
+    """
+    Aggregate values by given region.
+
+    Parameters
+    ----------
+    values
+        Values to aggregate.
+    by
+        Regions to aggregate by.
+    id_key
+        Key to group observations in `values` by. E.g. this could be transcript id for points.
+        Defaults to `FEATURE_KEY` for points, required for shapes.
+    value_key
+        Key to aggregate values by. This is the key in the values object.
+        If nothing is passed here, assumed to be a column of ones.
+        For points, this could be probe intensity.
```

Comment:
I don't understand this example.

Reply:
basically it's the column in poijnts that you want to aggregate by, so if it's genes, it would return the gene counts matrix with `agg_func = "counts"` whereas for a float, `agg_func = "mean"` could work./

Reply:
done